web-api/src/Kerbero.Infrastructure/SmartLockKeys/Repositories/SmartLockKeyPersistentRepository.cs

```diff
@@ -0,0 +1,53 @@
+using FluentResults;
+using Kerbero.Domain.Common.Errors;
+using Kerbero.Domain.SmartLockKeys.Entities;
+using Kerbero.Domain.SmartLockKeys.Repositories;
+using Kerbero.Infrastructure.Common.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Npgsql;
+
+namespace Kerbero.Infrastructure.SmartLockKeys.Repositories;
+
+public class SmartLockKeyPersistentRepository: ISmartLockKeyPersistentRepository
+{
+	private readonly ILogger<SmartLockKeyPersistentRepository> _logger;
+	private readonly IApplicationDbContext _applicationDbContext;
+
+	public SmartLockKeyPersistentRepository(ILogger<SmartLockKeyPersistentRepository> logger, IApplicationDbContext applicationDbContext)
+	{
+		_logger = logger;
+		_applicationDbContext = applicationDbContext;
+	}
+
+	public async Task<Result<SmartLockKey>> Create(SmartLockKey smartLockKey)
+	{
+		try
+		{
+			var smartLockKeyEntityEntry = _applicationDbContext.SmartLockKeys.Add(smartLockKey);
+			await _applicationDbContext.SaveChangesAsync();
+			return smartLockKeyEntityEntry.Entity;
```

Comment:
```suggestion
			return smartLockKeyEntityEntry.Entity;
```
why returning the same reference passed in? (to create the false sense that is different from the one passed in?)

Reply:
Because returns autogenerated values, like the id